src/OrbitGl/App.cpp

```diff
@@ -1835,15 +1842,17 @@ orbit_base::Future<ErrorMessageOr<std::filesystem::path>> OrbitApp::RetrieveModu
 orbit_base::Future<ErrorMessageOr<std::filesystem::path>> OrbitApp::RetrieveModuleWithDebugInfo(
     const std::string& module_path, const std::string& build_id) {
   auto loaded_module = RetrieveModule(module_path, build_id);
-  return loaded_module.ThenIfSuccess(
+  return loaded_module.Then(
       main_thread_executor_,
-      [this, module_path](
-          const std::filesystem::path& local_file_path) -> ErrorMessageOr<std::filesystem::path> {
-        auto elf_file = orbit_object_utils::CreateElfFile(local_file_path);
+      [this, module_path](const ErrorMessageOr<std::filesystem::path>& local_file_path)
+          -> ErrorMessageOr<std::filesystem::path> {
+        if (local_file_path.has_error()) return local_file_path;
+
+        auto elf_file = orbit_object_utils::CreateElfFile(local_file_path.value());
 
         if (elf_file.has_error()) return elf_file.error();
 
-        if (elf_file.value()->HasDebugInfo()) return local_file_path;
+        if (elf_file.value()->HasDebugInfo()) return local_file_path.value();
```

Comment:
Same here.. why touch this?

Reply:
Mainly to set the metric status code when `loaded_module` has error message.